src/traits.rs

```diff
@@ -198,16 +172,65 @@ pub trait Integer:
     }
 }
 
-/// Signed integers.
-pub trait Signed: Integer {
+/// Signed [`Integer`]s.
+pub trait Signed:
+    Div<NonZero<Self>, Output = CtOption<Self>>
+    + for<'a> Div<&'a NonZero<Self>, Output = CtOption<Self>>
+    + Integer
+{
     /// Corresponding unsigned integer type.
     type Unsigned: Unsigned;
+
+    /// The sign and magnitude of this [`Signed`].
+    fn abs_sign(&self) -> (Self::Unsigned, ConstChoice);
+
+    /// The magnitude of this [`Signed`].
+    fn abs(&self) -> Self::Unsigned {
+        self.abs_sign().0
+    }
+
+    /// Whether this [`Signed`] is negative, as a [`ConstChoice`].
+    fn is_negative(&self) -> ConstChoice;
+
+    /// Whether this [`Signed`] is positive, as a [`ConstChoice`].
+    fn is_positive(&self) -> ConstChoice;
```

Comment:
Used `ConstChoice` for all of these though that may not be the best option. I know elsewhere we have some hacks like using `ConstChoice` for one which is provided as a `const fn`, and another which is a trait method which returns `Choice`

(would that we didn't have to deal with such a split, sigh)